libs/git/view.go

```diff
@@ -72,8 +72,8 @@ func (v *View) IgnoreDirectory(dir string) (bool, error) {
 	return v.Ignore(dir + "/")
 }
 
-func NewView(root vfs.Path) (*View, error) {
-	repo, err := NewRepository(root)
+func NewView(worktreeRoot, root vfs.Path) (*View, error) {
```

Comment:
Even locally, worktreeRoot can be an outside root. Have not looked into this but we probably should construct the view solely on the worktree root. Here a unified interface with the `ignore()` method like proposed above can help.

Reply:
Not sure what you propose. Note, the intention here is to match previous functionality exactly: the main difference is that previously Repository would calculate worktree root and now it's passed explicitly. But the value is the same.